.circleci/config.yml

```diff
@@ -7,14 +7,42 @@ jobs:
       - checkout
       - run: git submodule sync --recursive
       - run: git submodule update --recursive --init
+
       - restore_cache:
-          key: _build
+          keys:
+            - v1-dependency-cache-{{ arch }}-{{ checksum "mix.lock" }}
+            - v1-dependency-cache-{{ arch }}
```

Comment:
Can you explain exactly what this accomplishes over the previous?

Reply:
@hayesgm I added cache fallbacks. so if ```v1-dependency-cache-{{ arch }}-{{ checksum "mix.lock" }}``` cache does not exist, it fallbacks to ```v1-dependency-cache-{{ arch }}``` and etc.

Recently I encountered a problem with elixir and circle ci on another project. Circle CI sometimes fails to find cache even if ```mix.lock``` (and its hash respectively) did not change. Caching is very critical for dialyzer check speed. Adding ```arch``` template key solves the problem.

https://discuss.circleci.com/t/use-the-arch-cache-template-key-if-you-rely-on-cached-compiled-binary-dependencies/16129